searx/engines/pixiv.py

```diff
@@ -44,13 +61,14 @@ def request(query, params):
 def response(resp):
     results = []
     data = resp.json()
+    working_proxies = check_proxies(Proxy)
```

Comment:
That will take a lot of time for each request and we don't test whether the proxy is working for engines like Piped or Invidious either, it's up to the instance maintainer to pick reliable proxies.

I think the changes with the latest commit should be undone.

Reply:
To check while engine is initialized, you can implement a [`init()`](https://docs.searxng.org/dev/engines/demo/demo_online.html) function in your engine module.

Here is by example what elasticsearch engine checks on initialization:

https://github.com/searxng/searxng/blob/11c0651ef47a72395d2d386d7600d951b0987ea0/searx/engines/elasticsearch.py#L59-L64